src/helpers/metrics.ts

```diff
@@ -1,12 +1,27 @@
 import client from 'prom-client';
 import promBundle from 'express-prom-bundle';
-import type { Express } from 'express';
+import type { Express, Request, Response } from 'express';
 import { size as queueSize } from '../lib/queue';
 import getModerationList from '../lib/moderationList';
+import DigitalOcean from '../lib/metrics/digitalOcean';
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import { rpcError } from './utils';
+
+const METRICS_AUTHORIZATION = process.env.METRICS_AUTHORIZATION || '';
+
+export const authChecker = async (req: Request, res: Response, next: any) => {
+  const { authorization = 'Bearer ' } = req.headers;
+
+  if (METRICS_AUTHORIZATION && authorization.split(' ')[1] !== METRICS_AUTHORIZATION) {
+    return rpcError(res, 'UNAUTHORIZED', '');
+  }
+  next();
+};
 
 export default function initMetrics(app: Express) {
   initCustomMetrics();
 
+  app.get('/metrics', authChecker);
```

Comment:
this does nothing, only checks the authorization header for `/metrics` endpoint. Is it supposed to be this way?

Reply:
When auth key check fail, this returns a 401 error, and stop processing all other routes, like a `break`

Reply:
I see... this route comes from `express-prom-bundle`

Reply:
Yes, it's a magic routes set by the middleware